### ISSUE TITLE
Spring hateoas integration: ensure compatibility with 1.4

### ISSUE THREAD

**Issue by ask4gilles** [Blazebit/blaze-persistence#1426]
Hi,

When trying to upgrade to spring boot `2.6.2`, I got this exception regarding the `spring-hateoas` integration:
```
Caused by: java.lang.NoSuchMethodError: 'void org.springframework.hateoas.EntityModel.<init>(java.lang.Object, org.springframework.hateoas.Link[])'
	at com.blazebit.persistence.spring.hateoas.webmvc.KeysetAwarePagedResourcesAssembler.lambda$toModel$0(KeysetAwarePagedResourcesAssembler.java:80)
	at com.blazebit.persistence.spring.hateoas.webmvc.KeysetAwarePagedResourcesAssembler.createModel(KeysetAwarePagedResourcesAssembler.java:133)
	at com.blazebit.persistence.spring.hateoas.webmvc.KeysetAwarePagedResourcesAssembler.toModel(KeysetAwarePagedResourcesAssembler.java:88)
	at com.blazebit.persistence.spring.hateoas.webmvc.KeysetAwarePagedResourcesAssembler.toModel(KeysetAwarePagedResourcesAssembler.java:80)
```
Could you please make the integration compatible with `spring-hateoas` 1.4 and the changes made by @odrotbohm ?
Thanks!

**Comment by odrotbohm** [comment]
Switching to the factory methods should do the trick. They had been deprecated for quite a while already.

**Comment by beikov** [comment]
Thanks for the report! Too bad you ran into this today when I just released a new version 😞 
If you had reported this earlier I would have been able to squeeze the fix in for that release. Is the update to the latest Spring HATEOAS very critical to you, or is it ok to wait e.g. 2 weeks? I'd like to wait a bit for feedback on this release before pushing out another one.

**Comment by ask4gilles** [comment]
Yeah, I had the same feeling when I saw the release announcement. No rush needed,take your time 😉

**Comment by beikov** [comment]
Fixed via 4092ddc6afad23dd2afb46c7f87cd0c5eb17898f
